datafusion/functions-nested/src/position.rs

```diff
@@ -172,13 +174,10 @@ fn general_position_dispatch<O: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<Ar
 
     // if `start_from` index is out of bounds, return error
     for (arr, &from) in list_array.iter().zip(arr_from.iter()) {
-        if let Some(arr) = arr {
-            if from < 0 || from as usize > arr.len() {
-                return internal_err!("start_from index out of bounds");
-            }
-        } else {
-            // We will get null if we got null in the array, so we don't need to check
-        }
+        assert_or_internal_err!(
```

Comment:
```suggestion
        // If `arr` is `None`: we will get null if we got null in the array, so we don't need to check
        assert_or_internal_err!(
```
I suggest we keep the comment.